src/App.jsx

```diff
@@ -14,45 +16,81 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_ALPHABET = 'alphabet';
+const SORT_FIELD_LENGTH = 'length';
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverseField, setReverseField] = useState(false);
+  const goods = [...goodsFromServer];
+
+  const onClickReverse = () => {
+    setReverseField(!reverseField);
+  };
+
+  if (sortField === SORT_FIELD_ALPHABET) {
+    goods.sort();
+  }
+
+  if (sortField === SORT_FIELD_LENGTH) {
+    goods.sort((firstGood, nextGood) => firstGood.length - nextGood.length);
+  }
+
+  if (reverseField) {
+    goods.reverse();
+  }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button is-info',
+            { 'is-light': sortField !== SORT_FIELD_ALPHABET })}
+          onClick={() => setSortField(SORT_FIELD_ALPHABET)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn('button is-success',
+            { 'is-light': sortField !== SORT_FIELD_LENGTH })}
+          onClick={() => setSortField(SORT_FIELD_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button is-warning',
+            { 'is-light': !reverseField })}
+          onClick={onClickReverse}
+        >
+          Reverse
+        </button>
+
+        {(sortField || reverseField)
+          && (
+            <button
+              type="button"
+              className="button is-danger is-light"
+              onClick={() => {
+                setSortField('');
+                setReverseField(false);
+              }}
```

Comment:
It's better to create separate functions to handle events. This way, your code will be easier to maintain and change
```suggestion
              onClick={onReset}
```